### ISSUE TITLE
quote-operators don't work with utf8-delimiters

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5873]
Migrated from [rt.perl.org#16823](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=16823) (status was 'resolved')

Searchable as RT16823$

**Comment by p5pRT** [comment]
### From pcg@goof.com

#### Created by root@cerebro.laendle




It seems that the "use utf8" pragma does not work with the quote\-like
operators\, i\.e\.

&nbsp;   q ü test ü;
&nbsp;   q· test ·;

Result in "Unrecognized character \\xBC at x\.pl line 1\." and "Unrecognized
character \\xB7 at x\.pl line 1\."\, respectively\, rather than just work as I
would expect\.

Ah\, I'd soo much like to use this great way of quoting perl source ;\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.8.0:

Configured by root at Sat Jul 20 01:25:22 CEST 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4, archname=i686-linux
    uname='linux cerebro 2.4.18-pre8-ac3 #2 smp tue feb 5 17:35:23 cet 2002 i686 unknown '
    config_args=''
    hint=previous, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='gcc-2.95.4', ccflags ='-I/opt/include -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-Os -funroll-loops -mcpu=pentium -march=pentium -g',
    cppflags='-I/opt/include -D_GNU_SOURCE -I/opt/include -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/opt/include -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    ccversion='', gccversion='2.95.4 20010319 (prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc-2.95.4', ldflags =''
    libpth=/usr/lib /opt/lib
    libs=-lcrypt -ldl -lm -lc
    perllibs=-lcrypt -ldl -lm -lc
    libc=/lib/libc-2.2.5.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.2.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.0:
    /root/src/sex
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    .


Environment for perl v5.8.0:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=de_DE@euro
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/root/s2:/root/s:/opt/qt/bin:/bin:/usr/bin:/usr/app/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/usr/app/bin:/usr/app/sbin:/usr/X11/bin:/opt/j2sdk1.4.0_01//bin:/opt/bin:/opt/sbin:.:/root/cc/dejagnu/bin
    PERL5LIB=/root/src/sex
    PERLDB_OPTS=ornaments=0
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @rgarcia

Marc Lehmann \(via RT\) \<perlbug@&#8203;perl\.org> wrote&#8203;:
> 
> It seems that the "use utf8" pragma does not work with the quote\-like
> operators\, i\.e\.
> 
>    q Ã¼ test Ã¼;
>    qÂ· test Â·;
> 
> Result in "Unrecognized character \\xBC at x\.pl line 1\." and "Unrecognized
> character \\xB7 at x\.pl line 1\."\, respectively\, rather than just work as I
> would expect\.

A quick look at this indicates that the string is correctly parsed\, but
the tokenizer leaves the last byte of the terminating delimiter
in the read buffer\, hence the error\.

How do I know how many bytes I should skip in order to position correctly
the buffer pointer to the 1st character after the multi\-byte terminating
delimiter ?

**Comment by p5pRT** [comment]
### From @nwc10

On Thu\, Aug 29\, 2002 at 10&#8203;:32&#8203;:13AM \+0200\, Rafael Garcia\-Suarez wrote&#8203;:
> Marc Lehmann \(via RT\) \<perlbug@&#8203;perl\.org> wrote&#8203;:
> > 
> > It seems that the "use utf8" pragma does not work with the quote\-like
> > operators\, i\.e\.
> > 
> >    q Ã¼ test Ã¼;
> >    qÂ· test Â·;
> > 
> > Result in "Unrecognized character \\xBC at x\.pl line 1\." and "Unrecognized
> > character \\xB7 at x\.pl line 1\."\, respectively\, rather than just work as I
> > would expect\.
> 
> A quick look at this indicates that the string is correctly parsed\, but
> the tokenizer leaves the last byte of the terminating delimiter
> in the read buffer\, hence the error\.

Are you sure it's the last byte? Instead of "all but the first"
\(ie what will it do with characters that become 3 or more bytes in utf8\)

> How do I know how many bytes I should skip in order to position correctly
> the buffer pointer to the 1st character after the multi\-byte terminating
> delimiter ?

presumably\, as the delimiter is 1 character\, in utf8 mode if the pointer is
left starting on the start of a utf8 sequence\, then it should advance byte by
byte until it hits the first non\-continuation byte\.
\(and if it's actually parsed starting on the byte after the first byte
\(ie the right place for 7 or 8 bit data\) then I think the robust way is to
back up one\, see if the previous byte is a valid utf8 start byte\, and if
so advance through continuation bytes\. Mainly because the use 'utf8';
changes to the tokenizer seem to be a bit of a hack \- in 'utf8' mode the
tokenizer is checking each apparently utf8 sequence\, and if the tokenizer
finds the sequence to be malformed it quietly accepts it as 8 bit bytes\,
rather than raising an error about malformed input\)

This message comes without the "benefit" of looking at the tokenizer
source for well over 24 hours\, so should could be as wrong as the average
Usenet message\.

Nicholas Clark

**Comment by p5pRT** [comment]
### From @rgarcia

Nicholas Clark \<nick@&#8203;ccl4\.org> wrote&#8203;:
> On Thu\, Aug 29\, 2002 at 10&#8203;:32&#8203;:13AM \+0200\, Rafael Garcia\-Suarez wrote&#8203;:
> > A quick look at this indicates that the string is correctly parsed\, but
> > the tokenizer leaves the last byte of the terminating delimiter
> > in the read buffer\, hence the error\.
> 
> Are you sure it's the last byte? Instead of "all but the first"
> \(ie what will it do with characters that become 3 or more bytes in utf8\)

Yes\, it's the second byte\.

> > How do I know how many bytes I should skip in order to position correctly
> > the buffer pointer to the 1st character after the multi\-byte terminating
> > delimiter ?
> 
> presumably\, as the delimiter is 1 character\, in utf8 mode if the pointer is
> left starting on the start of a utf8 sequence\, then it should advance byte by
> byte until it hits the first non\-continuation byte\.

Well\, currently the delimiter is one byte\. Getting utf\-8 delimiters to
work means including some support for multi\-byte delimiters\. \(A similar
problem arises also with the 1st delimiter\.\) I'm not familiar with perl's
utf\-8 API \(and I don't even know if it's worth the pain\)\.

**Comment by p5pRT** [comment]
### From nick.ing-simmons@elixent.com

Rafael Garcia\-Suarez \<raphel\.garcia\-suarez@&#8203;hexaflux\.com> writes&#8203;:
>Marc Lehmann \(via RT\) \<perlbug@&#8203;perl\.org> wrote&#8203;:
>> 
>> It seems that the "use utf8" pragma does not work with the quote\-like
>> operators\, i\.e\.
>> 
>>    q Ã¼ test Ã¼;
>>    qÂ· test Â·;
>> 
>> Result in "Unrecognized character \\xBC at x\.pl line 1\." and "Unrecognized
>> character \\xB7 at x\.pl line 1\."\, respectively\, rather than just work as I
>> would expect\.
>
>A quick look at this indicates that the string is correctly parsed\, but
>the tokenizer leaves the last byte of the terminating delimiter
>in the read buffer\, hence the error\.
>
>How do I know how many bytes I should skip in order to position correctly
>the buffer pointer to the 1st character after the multi\-byte terminating
>delimiter ?

Usually something like&#8203;:

s \+= UTF8SKIP\(s\);


\-\- 
Nick Ing\-Simmons
http&#8203;://www\.ni\-s\.u\-net\.com/

**Comment by p5pRT** [comment]
### From @jhi

Patch attached\.

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @jhi

[16823.pat.gz](https://rt-archive.perl.org/perl5/Ticket/Attachment/43616/34702/16823.pat.gz)

**Comment by p5pRT** [comment]
### From @jhi

Stratch that patch\.  It works \-\- but only for the examples in the bug
report\, that is\, only for Latin\-1 bytes encoded in UTF\-8\.  I'm pretty
certain it doesn't work for characters 0x100 and past\.  I'll need to
munge the S\_scan\_str code a bit more to able to do that\.  Also\, the
patch only works for use utf8\, not for use encoding 'utf8'\.  Getting
that will again require a bit more work\.

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @jhi

> Stratch that patch\.  It works \-\- but only for the examples in the bug
> report\, that is\, only for Latin\-1 bytes encoded in UTF\-8\.  I'm pretty
> certain it doesn't work for characters 0x100 and past\.  I'll need to
> munge the S\_scan\_str code a bit more to able to do that\.  Also\, the
> patch only works for use utf8\, not for use encoding 'utf8'\.  Getting
> that will again require a bit more work\.

It's even worse than I thought\, especially with the encoding pragma\.

Firstly\, the whole S\_scan\_str\(\) is still very much in the 'one byte
per character' world\, it's full of \*s and s\+\+ and s\[1\] and s\+1\.
I think it could be made to work with just the 'use utf8' by using
enough utf8\_to\_uvchr\(\(U8\*\)s\,0\) and UTF8SKIP\(s\)\.\.\.

Secondly\, the encoding pragma poses an evil problem\.  When the
S\_scan\_str\(\) is entered\, we have just seen the q/qq/m/s/tr/\.\.\.\, but we
must \(skip the optional space\) and then extract the starting quoting
character\, and then look for the ending quoting character\.  The
problem is that when we enter\, the recoding as dictated by the
encoding pragma hasn't been done yet \(it's now done in S\_scan\_str\(\)
\*after\* the ending quoting character has been read\)\.  So we may have
some variable length encoding that requires us to read several bytes
to get the starting quoting character\.  So basically I think we would
have to do the recoding before doing the quote scanning\-\- which means
a new completely different string than we currently having in the
parsing buffer\.  Pain\. Pain\. Pain\.  I must say that the required
effort for the gained benefit does seem a bit unbalanced\.

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @nwc10

On Sun\, Dec 01\, 2002 at 06&#8203;:01&#8203;:51PM \+0200\, Jarkko Hietaniemi wrote&#8203;:

> It's even worse than I thought\, especially with the encoding pragma\.
> 
> Firstly\, the whole S\_scan\_str\(\) is still very much in the 'one byte
> per character' world\, it's full of \*s and s\+\+ and s\[1\] and s\+1\.
> I think it could be made to work with just the 'use utf8' by using
> enough utf8\_to\_uvchr\(\(U8\*\)s\,0\) and UTF8SKIP\(s\)\.\.\.

> Secondly\, the encoding pragma poses an evil problem\.  When the
> S\_scan\_str\(\) is entered\, we have just seen the q/qq/m/s/tr/\.\.\.\, but we
> must \(skip the optional space\) and then extract the starting quoting
> character\, and then look for the ending quoting character\.  The
> problem is that when we enter\, the recoding as dictated by the
> encoding pragma hasn't been done yet \(it's now done in S\_scan\_str\(\)
> \*after\* the ending quoting character has been read\)\.  So we may have
> some variable length encoding that requires us to read several bytes
> to get the starting quoting character\.  So basically I think we would
> have to do the recoding before doing the quote scanning\-\- which means
> a new completely different string than we currently having in the
> parsing buffer\.  Pain\. Pain\. Pain\.  I must say that the required
> effort for the gained benefit does seem a bit unbalanced\.

Erk\. But how do we know how much to recode? Presumably we really need to
re\-write S\_scan\_str with \(say\) shiftjs\_to\_uvchr and SHIFTJSSKIP so that
we advance 1 \(say\) Shift JS encoded character at a time until we meet
the terminator character again\.

I can't see a good answer to this\. It feels like we need a delimiter scanning
routine \(at least one good enough to find the end delimiter\) for every
encoding we allow strings in\.

Presumably a use Encoding 'posix\-bc' would mean that I'd write

qqZÈÅÓÓÖk@&#8203;æÖÙÓÄZZ

and somehow the scan\_str has to know that I was writing qq\!HELLO\, WORLD\\\!\!
with a \! escaped with a \\

\(I hope I code that EBCDIC/ASCII encoding the correct way round\)

Nicholas Clark
\-\- 
INTERCAL better than perl?    http&#8203;://www\.perl\.org/advocacy/spoofathon/

**Comment by p5pRT** [comment]
### From @jhi

> Erk\. But how do we know how much to recode? Presumably we really need to
> re\-write S\_scan\_str with \(say\) shiftjs\_to\_uvchr and SHIFTJSSKIP so that
> we advance 1 \(say\) Shift JS encoded character at a time until we meet
> the terminator character again\.

Well\, I don't think that's too much different from the problem of
first reading '\.\.\. and then having to read in another block from the
disk to finish off the quoted string\.  There's no need for
shiftjis\_to\_uvchr\(\) and SHIFTJSSKIP since once the recoding is done
the characters\, whatever they were\, have been converted to UTF\-8
encoded Unicode\.  Besides\, there are dozens of encodings\, so having
an API for each of them would be Madness\.

> I can't see a good answer to this\. It feels like we need a delimiter scanning
> routine \(at least one good enough to find the end delimiter\) for every
> encoding we allow strings in\.

See above\.  What is most needed for the above is a lightweight API for
"read in enough bytes \(in whatever encoding\) to read in one character
\(and convert to UTF\-8 while you are doing the reading\)"\.  I \*think\*
currently the PerlIO&#8203;::encoding API is a bit too heavy to be called
repeatedly&#8203;: it works okay for converting whole chunks\, like string
constants\.

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From pcg@goof.com

On Sun\, Dec 01\, 2002 at 04&#8203;:02&#8203;:19PM \-0000\, Jarkko Hietaniemi \<perlbug@&#8203;perl\.org> wrote&#8203;:
> It's even worse than I thought\, especially with the encoding pragma\.

Even if it's obvious\, I think it's great to care for such a detail\. How about
this solution&#8203;:

&nbsp;   use utf8 => works

&nbsp;   everything else => documented not to work\.

Sourcecode not in utf\-8 but multibyte\-encoding is rare currently\, and
supporting a very common \(or standard\) case like utf8\.

There are worse problems\, like switching encoding in the middle of strings
etc\.\. And maybe we can at one point use utf\-8 characters as new operators\,
finalyl despecialiazing ascii ;\)

Well\.\.\. in any case\, \_I\_ would be happy to have this work for utf\-8 only\,
and I don't think too many people would miss that feature\.

\(Even I can live without it\, in fact\)

> parsing buffer\.  Pain\. Pain\. Pain\.  I must say that the required
> effort for the gained benefit does seem a bit unbalanced\.

I agree\, if that helps ;\)

\-\- 
&nbsp;      \-\-\-\-\-==\-                                             |
&nbsp;      \-\-\-\-==\-\- \_                                           |
&nbsp;      \-\-\-==\-\-\-\(\_\)\_\_  \_\_ \_\_\_\_  \_\_       Marc Lehmann      \+\-\-
&nbsp;      \-\-==\-\-\-/ / \_ \\/ // /\\ \\/ /       pcg@&#8203;goof\.com      |e|
&nbsp;      \-=====/\_/\_//\_/\\\_\,\_/ /\_/\\\_\\       XX11\-RIPE         \-\-\+
&nbsp;    The choice of a GNU generation                       |
&nbsp;                                                         |

**Comment by p5pRT** [comment]
### From goldbb2@earthlink.net

Jarkko Hietaniemi wrote&#8203;:
> 
> > Erk\. But how do we know how much to recode? Presumably we really
> > need to re\-write S\_scan\_str with \(say\) shiftjs\_to\_uvchr and
> > SHIFTJSSKIP so that we advance 1 \(say\) Shift JS encoded character at
> > a time until we meet the terminator character again\.
> 
> Well\, I don't think that's too much different from the problem of
> first reading '\.\.\. and then having to read in another block from the
> disk to finish off the quoted string\.  There's no need for
> shiftjis\_to\_uvchr\(\) and SHIFTJSSKIP since once the recoding is done
> the characters\, whatever they were\, have been converted to UTF\-8
> encoded Unicode\.  Besides\, there are dozens of encodings\, so having
> an API for each of them would be Madness\.
> 
> > I can't see a good answer to this\. It feels like we need a delimiter
> > scanning routine \(at least one good enough to find the end
> > delimiter\) for every encoding we allow strings in\.
> 
> See above\.  What is most needed for the above is a lightweight API for
> "read in enough bytes \(in whatever encoding\) to read in one character
> \(and convert to UTF\-8 while you are doing the reading\)"\.  I \*think\*
> currently the PerlIO&#8203;::encoding API is a bit too heavy to be called
> repeatedly&#8203;: it works okay for converting whole chunks\, like string
> constants\.

What's wrong with local\(\)ly setting $/ to the \(possibly utf8\) delimiter?

\-\- 
my $n = 2; print \+\(split //\, 'e\,4c3H r ktulrnsJ2tPaeh'
\.\.\.\."\\n1oa\! er"\)\[map $n = \($n \* 24 \+ 30\) % 31\, \(42\) x 26\]

**Comment by p5pRT** [comment]
### From @jhi

> What's wrong with local\(\)ly setting $/ to the \(possibly utf8\) delimiter?

Huh?

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From goldbb2@earthlink.net

Jarkko Hietaniemi wrote&#8203;:
> 
> > What's wrong with local\(\)ly setting $/ to the \(possibly utf8\)
> > delimiter?
> 
> Huh?

Oops\, nevermind\, I was confused\.

\-\- 
my $n = 2; print \+\(split //\, 'e\,4c3H r ktulrnsJ2tPaeh'
\.\.\.\."\\n1oa\! er"\)\[map $n = \($n \* 24 \+ 30\) % 31\, \(42\) x 26\]

**Comment by p5pRT** [comment]
### From inaba@st.rim.or.jp

Dan Kogai wrote&#8203;:

> Porters\,
>
>    In the recent discussion in various perl\-related MLs in Japanese\, I
> have discovered a problem that the encoding pragma does not work on
> such multibyte encodings as Shift\_JIS which uses 0x00\-0x7f ranges in
> the 2nd byte\.  Though not test I am pretty sure big5 is also prone to
> this\.

\<Skip sample script hex dump>

>    The perl script is a valid perl script in Shift JIS but the quoted
> character \(U\+80fd\, \\x94\\x5c in Shift\_JIS\) uses \\x5c in the 2nd byte\,
> mangling the script\.  The encoding pragma needs to be parsable
> ASCII\-wise\.
>    Fortunately\, the encoding pragma offers a different approach via
> Filter=>1\. \.\.\.

Attached patch\(for breadperl @&#8203;18609\) is an attempt to fix this problem
without
Filter=>1 option\.

It does&#8203;:
&nbsp;  \- Modify method\_decode \(Encode/Encode\.xs\) and do\_encode
\(Encode/encengine\.c\) to take
&nbsp;    terminator argument
&nbsp;  \- Add a method cat\_decode to Encoding object which take destination\,
source\,
&nbsp;    offset and terminator as arguments\.
&nbsp;    \(Implemented packages are&#8203;: Encode&#8203;::XS\, Encode&#8203;::utf8 and
Encode&#8203;::JP&#8203;::JIS7\)
&nbsp;  \- Add a function sv\_cat\_decode\(\) to append decoded UTF8 string with offset
and
&nbsp;    terminator using method cat\_decode\.
&nbsp;  \- When scan\_str\(\)  parses input with PL\_encoding\, use sv\_cat\_decode\(\) with
PL\_linestr\+
&nbsp;    offset and specified terminator\.
&nbsp;  \-  In fact\, I have started to make this patch for
&nbsp;         Subject&#8203;: Re&#8203;: \[PATCH\] \[perl \#16823\] quote\-operators don't work with
utf8\-delimiters
&nbsp;         Date&#8203;: Sun\, 1 Dec 2002 18&#8203;:01&#8203;:51 \+0200
&nbsp;         From&#8203;:Jarkko Hietaniemi \<jhi@&#8203;iki\.fi>
&nbsp;     So parsing under \`use utf8' is also changed in scan\_str\(\)\.

 Though not concerns the main intent\, modifies sv\_recode\_to\_utf8\(\) to
&nbsp;    \- Change \!DO\_UTF8\(sv\) to \!Sv\_UTF8\(sv\) && \!IN\_BYTES
&nbsp;    \- Add save\_re\_context\(\)
&nbsp;    \- Retract my useless code which checks UTF8\_IS\_INVARIANT
\-\-
&nbsp;    Inaba Hiroto    \<inaba@&#8203;st\.rim\.or\.jp>

**Comment by p5pRT** [comment]
### From inaba@st.rim.or.jp

[enc-patch.gz](https://rt-archive.perl.org/perl5/Ticket/Attachment/50018/38553/enc-patch.gz)

**Comment by p5pRT** [comment]
### From @jhi

Thanks\, applied now as change \#18660\.  Hopefully Dan can easily
merge this into the CPAN Encode\.

\-\- 
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> http&#8203;://www\.iki\.fi/jhi/ "There is this special
biologist word we use for 'stable'\.  It is 'dead'\." \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From dankogai@dan.co.jp

On Thursday\, February 6\, 2003\, at 03&#8203;:25 AM\, Jarkko Hietaniemi wrote&#8203;:
> Thanks\, applied now as change \#18660\.  Hopefully Dan can easily
> merge this into the CPAN Encode\.

Still working on it besides the bug report \(and its fix\) I just 
reported\.\.\.\.

Dan

**Comment by p5pRT** [comment]
### From dankogai@dan.co.jp

On Thursday\, February 6\, 2003\, at 03&#8203;:40 AM\, Dan Kogai wrote&#8203;:
> On Thursday\, February 6\, 2003\, at 03&#8203;:25 AM\, Jarkko Hietaniemi wrote&#8203;:
>> Thanks\, applied now as change \#18660\.  Hopefully Dan can easily
>> merge this into the CPAN Encode\.
>
> Still working on it besides the bug report \(and its fix\) I just 
> reported\.\.\.\.

Good news\.  Inaba\-san's patch does no harm to perl 5\.8\.0\.  I'll just go 
ahead merge it;  All I have to do is to replace those that are patched\.

Dan

**Comment by p5pRT** [comment]
### From @jhi

I think the original issue got resolved \(UTF\-8 delimiters under use utf8 seem to work
now\)\, so I'm marking the problem ticket as resolved\.  \(There's still "fun" to be had
with "use encoding"\, but let someone open up a new ticket for that\.\)

**Comment by p5pRT** [comment]
@jhi - Status changed from 'new' to 'resolved'
